zerver/actions/message_edit.py

```diff
@@ -1003,6 +1031,13 @@ def user_info(um: UserMessage) -> Dict[str, Any]:
                     visibility_policy=new_visibility_policy,
                 )
 
+    if topic_name is not None and not target_topic_has_messages:
+        first_message_moved = changed_messages.first()
```

Comment:
I think we can directly use `target_message` here ? -- This will avoid the extra DB query too.

Reply:
if we use `target_message` directly, I guess there would be a small bug -- when you select "move message" for some message which is not the first message of the topic, and you choose the option to move all messages of topic, then the `automatic follow` is applied to this `target_message` on which we selected move, rather than the first message of the topic.

Reply:
I updated it according to your point.